splice_machine/src/main/java/com/splicemachine/derby/impl/stats/StoreCostControllerImpl.java

```diff
@@ -235,12 +235,12 @@ else if (nonNullCount == 0)
 
     @Override
     public long cardinality(int columnNumber) {
-        if (missingPartitions > 0) {
```

Comment:
@dgomezferro, I'm not completely sure, but the previous behavior looks like a copy-paste typo. In the same code module StoreCostControllerImpl.java, we have the following behavior for rowCount and notNullCount when missingPartitions is not 0:
    @Override
    public double rowCount() {
        if (missingPartitions > 0) {
            assert tableStatistics.numPartitions() > 0: "Number of partitions cannot be 0 ";
            return tableStatistics.rowCount() + tableStatistics.rowCount() * (missingPartitions / tableStatistics.numPartitions());
        }
        else
            return tableStatistics.rowCount();
    }

    @Override
    public double nonNullCount(int columnNumber) {
        if (missingPartitions > 0) {
            assert tableStatistics.numPartitions() > 0: "Number of partitions cannot be 0";
            return tableStatistics.notNullCount(columnNumber - 1) + tableStatistics.notNullCount(columnNumber - 1) * (missingPartitions / tableStatistics.numPartitions());
        } else
            return tableStatistics.notNullCount(columnNumber-1);
    }

As you can see, both use the known partition stats(either rowCount or notNullCount) to extrapolate. The behavior for cardinality looks like to be copied from notNullCount() without correction.

Reason for the current fix is that, unlike rowCount and notNullCount, there could be overlaps for unique values among different partitions, so I pick the behavior similar to the estimation of minValue() and maxValue() in the same module, that is, use the existing stats directly.